src/Transform/ONNX/ONNXRewrite.td

```diff
@@ -93,4 +102,39 @@ def MaxPoolSingleOutOpPaddingPattern: Pat<
   [(HasNonZeroInArrayAttr:$pads)]
 >;
 
+//===----------------------------------------------------------------------===//
+// Rewrite:
+// %0 = onnx.AveragePoolOp(%D : tensor<DShape>)
+//     {pads = [b0, b1, ... bK, e0, e1, ..., eK]} ->
+//         tensor<OutShape>
+//
+// as:
+// %0 = onnx.PadConstantValuePadOp(%D)
+//     {pads = [0, 0, b0, b1, ... bK, 0, 0, e0, e1, ..., eK]} ->
+//     tensor<DPaddedShape>
+// %1 = onnx.AveragePool(%0 : tensor<DPaddedShape>) {pads = [0, ..., 0]} ->
+//     tensor<OutShape>
+//===----------------------------------------------------------------------===//
+
+// Pad values are NaNs if count_include_pad is disabled (default).
```

Comment:
I have a high level question: if you do an average with count_include_pad set to false. Do you adjust the denominator?

e.g. avg(4,6, pad, pad) = (4+6)/2 5?

Reply:
Yes, I just adjust the denominator. NaN is signal to know a value is pad or not. By default `count_include_pad` is false. Please look at e2e tests in `test/backend/test.py` where passed e2e tests for AveragePool were added. So far all e2e tests for AveragePool passed.